text/0060-refashion-updates-apis.md

```diff
@@ -0,0 +1,429 @@
+- Title: Refashion Updates APIs
+- Start Date: 2021-08-13
+- Specification PR: [#55](https://github.com/meilisearch/specifications/pull/60)
+- Discovery Issue: [#48](https://github.com/meilisearch/product/issues/48)
+
+# Refashion Updates APIs
+
+## 1. Functional Specification
+
+### I. Summary
+
+The term `update` is not the best choice as it can be confused with document updates or `settings`. We have chosen to replace this term with `task`, which is more generic and better reflects the meaning of this API resource.
+
+As an additional change, we have reworked the format of an update to make it more in line with our expectations of an API that is supposed to be easily understandable and developer-oriented.
+
+The changes we will make to the response format of the `task` object lists will make it easier to add more functionality in a future iteration. See the Future Possibilities section for a brief overview.
+
+Two new API endpoints are added. Although quite simple, they allow to consult the list of tasks or a specific task without being forced to know the related index.
+
+#### Summary Key Points
+
+- The `update` resource is renamed `task`. The names of existing API routes are also changed to reflect this change.
+- Tasks are now also accessible as an independent resource of an index. `GET - /tasks`; `GET - /tasks/:taskUid`
+- The task `uid` is not incremented by index anymore. The sequence is generated globally.
+- A `task_not_found` error is introduced.
+- The format of the `task` object is updated.
+    - `updateId` becomes `uid`.
+    - Attributes of an error appearing in a `failed` `task` are now contained in a dedicated `error` object.
+    - `type` is no longer an object. It now becomes a string containing the values of its `name` field previously defined in the `type` object.
+    - The possible values for the `type` field are reworked to be more clear and consistent with our naming rules.
+    - A `details` object is added to contain specific information related to a `task` payload that was previously displayed in the `type` nested object. Previous `number` key is renamed `numberOfDocuments`.
```

Comment:
Out of curiosity, Is it planned that  `numberOfDocuments` returns in enqueued/processingfailed status ?

Reply:
This is only possible for tasks completed today because of the architecture of the update store. This is something that could be great in the future!